train_hierarchialXC.py

```diff
@@ -15,6 +15,18 @@
 import numpy as np
 import torch.nn.functional as F
 
+def generate_encodings(data_loader, ae):
+    vals = []
+
+    for x, _ in iter(data_loader):
+        vals.append(ae(x.float()))
+    vals = torch.cat(vals, dim=0)
+
+    vals = vals.detach().numpy()
+    vals = np.squeeze(vals, 1)
```

Comment:
Why squeeze?

Reply:
dimensions of vals is `[no_of_datapoints,1,1836]`

Reply:
That’s funny. I thought I fixed this.

If so, let it be like this.

Reply:
Or rather squeeze the inputs to be `batch size x dims`. Do this, this is better.

Reply:
IIRC, this issue will not happen if you use the data loader. I remember fixing the logic in the `__getitem__` method. So the squeeze should not be necessary. Consult the code in train_neuralXC for this

Reply:
Since this is for vals, I didn't quite get how do you plan to use data_loader for this

Reply:
`vals` is the concatenation of forward passes over the inputs to the network, which are items in the data loader.